### ISSUE TITLE
--sub-ass-style-overrides not applied when --sub-ass-override is set to force

### ISSUE THREAD

**Issue by low-batt** [mpv-player/mpv#14800]
### mpv Information

```bash
mpv v0.38.0-744-gdb19dbca5c-dirty Copyright © 2000-2024 mpv/MPlayer/mplayer2 projects
 built on Sep  4 2024 18:15:34
libplacebo version: v7.349.0 (v7.349.0-9-gefb89342)
FFmpeg version: n7.0.2
FFmpeg library versions:
   libavcodec      61.3.100
   libavdevice     61.1.100
   libavfilter     10.1.100
   libavformat     61.1.100
   libavutil       59.8.100
   libswresample   5.1.100
   libswscale      8.1.100
```

### Other Information

```text
- macOS version: macOS 13.6.9 (22G830)
- Source of mpv: Local build using `mpv-build` of `master` @ db19dbca5cd2fce2189005ca1ebbfd4e0467efae
- Introduced in version: unknown
```

### Reproduction Steps

- Download the attached `big_buck_bunny.ass.txt` file
- Rename it to `big_buck_bunny.ass`
- Copy some video to `big_buck_bunny.mp4` to match up with subtitle file
- Play the video with `--sub-ass-style-overrides=FontName=Wingdings --sub-ass-override=scale`
- Play the video with `--sub-ass-style-overrides=FontName=Wingdings --sub-ass-override=force`
- Notice the font is not applied with `--sub-ass-override` set to `force`

### Expected Behavior

I'm expecting that [sub-ass-style-overrides](https://mpv.io/manual/master/#options-sub-ass-style-overrides) overrides ASS style when [sub-ass-override](https://mpv.io/manual/stable/#options-sub-ass-override) is set to anything other than `no`.

From the mpv manual entry for the [sub-ass-override](https://mpv.io/manual/stable/#options-sub-ass-override) option:
> - **no**:	Render subtitles as specified by the subtitle scripts, without overrides.
> - **yes**:	Apply all the `--sub-ass-*` style override options. Changing the default for any of these options can lead to incorrect subtitle rendering (default).
> - **scale**:	Like yes, but also apply `--sub-scale`.
> - **force**:	Like yes, but also force all `--sub-*` options. Can break rendering easily.
> - **strip**:	Radically strip all ASS tags and styles from the subtitle. 

### Actual Behavior

With `--sub-ass-override` set to `yes` or `scale`  `--sub-ass-style-overrides` overrides ASS style. But when  `--sub-ass-override` is set to `force` or `strip`  `--sub-ass-style-overrides` is ignored.

With  `--sub-ass-override=no`:
```text
mpv/build/mpv --no-config --autofit=2048 --pause --start=1:08 --sub-ass-style-overrides=FontName=Wingdings --sub-ass-line-spacing=100 --sub-ass-override=no ~/Movies/big_buck_bunny.mp4
```
![no](https://github.com/user-attachments/assets/07fe482e-e3bb-4b28-8bfa-a0748b999121)

With  `--sub-ass-override=yes` the font override is applied as well as the line spacing:
```text
mpv/build/mpv --no-config --autofit=2048 --pause --start=1:08 --sub-ass-style-overrides=FontName=Wingdings --sub-ass-line-spacing=100 --sub-ass-override=yes ~/Movies/big_buck_bunny.mp4
```
![yes](https://github.com/user-attachments/assets/28a25f24-0d05-4114-abd6-d01491d324fe)

With `--sub-ass-override=scale` it looks just like `--sub-ass-override=yes`.

With  `--sub-ass-override=force` the font override is _not applied_, the line spacing is still overridden:
```text
mpv/build/mpv --no-config --gpu-debug  --log-file=output.txt --autofit=2048 --pause --start=1:08 --sub-ass-style-overrides=FontName=Wingdings --sub-ass-line-spacing=100 --sub-ass-override=force ~/Movies/big_buck_bunny.mp4
```
![force](https://github.com/user-attachments/assets/2d4f4991-3018-40cd-9349-9f9ebadd981c)

With  `--sub-ass-override=strip` the font override is _not applied_, the line spacing is still overridden:
```text
mpv/build/mpv --no-config --autofit=2048 --pause --start=1:08 --sub-ass-style-overrides=FontName=Wingdings --sub-ass-line-spacing=100 --sub-ass-override=strip ~/Movies/big_buck_bunny.mp4
```
![strip](https://github.com/user-attachments/assets/c3164752-fac6-40dc-990d-92a2d00bfab7)

### Log File

[output.txt](https://github.com/user-attachments/files/16882538/output.txt)

### Sample Files

[big_buck_bunny.ass.txt](https://github.com/user-attachments/files/16882541/big_buck_bunny.ass.txt)

### I carefully read all instruction and confirm that I did the following:

- [X] I tested with the latest mpv version to validate that the issue is not already fixed.
- [X] I provided all required information including system and mpv version.
- [X] I produced the log file with the exact same set of files, parameters, and conditions used in "Reproduction Steps", with the addition of `--log-file=output.txt`.
- [X] I produced the log file while the behaviors described in "Actual Behavior" were actively observed.
- [X] I attached the full, untruncated log file.
- [X] I attached the backtrace in the case of a crash.

**Comment by hooke007** [comment]
```
Like yes, but also force all --sub-* options. Can break rendering easily.
```

Accroding to manual, with `--sub-ass-override=force`, the actual font should be `--sub-font`.
I don't think `sub-ass-style-overrides` should work in this case.

**Comment by low-batt** [comment]
But "yes" means:
> Apply all the `--sub-ass-*` style override options

So if we aren't applying a `--sub-ass-*`  style override then it is not "Like yes". At least that is the way I read it. I was wondering what would take precedence if `--sub-font` was specified. My guess was that `--sub-ass-style-overrides` would be preferred over `--sub-font` as `--sub-ass*` are specific to ASS subtitles. But in this case `--sub-font` was not specified, so I'm really surprised the font specified by `--sub-ass-style-overrides` was ignored. That seems odd to me. If nothing else I'm thinking the manual needs to say something about this case.

**Comment by hooke007** [comment]
>  But in this case --sub-font was not specified, 

No, it was always specified.
https://mpv.io/manual/master/#options-sub-font

**Comment by low-batt** [comment]
Not specified as in I did not pass `--sub-font` on the command line.

**Comment by hooke007** [comment]
Any option has a default value.

**Comment by low-batt** [comment]
The behavior that caught me by surprise is that the `--sub-*` options override not just the styles in the ASS subtitle, but that they override the style specified in `--sub-ass-*` options. In reading the documentation for the  [sub-ass-override](https://mpv.io/manual/stable/#options-sub-ass-override) levels it feels additive. So I was very surprised when raising the level to `force` removed a style specified not in the subtitle, but in an ASS specific option.

**Comment by llyyr** [comment]
FWIW, you're reading the "stable" manual. I changed it on master. https://mpv.io/manual/master/#options-sub-ass-override

Also, I think specifying a style via `sub-ass-styles` should still work with `=force`

**Comment by hooke007** [comment]
> Also, I think specifying a style via sub-ass-styles should still work with =force

I don't agree. Which would made users more confused why sub-* doesn't work under `force` as expected.

**Comment by hooke007** [comment]
> Upon testing, it doesn't work anyway

Which one? Quickily test here `sub-font-size` work for me.

**Comment by llyyr** [comment]
> Which one?

sub-ass-styles doesn't work when override is set to force.

**Comment by hooke007** [comment]
> sub-ass-styles doesn't work when override is set to force.

As my first view, which is expected.

**Comment by low-batt** [comment]
I'm feeling rather bad about making a big deal about this particular behavior. I am good with whatever the mpv team decides to do about this issue including just closing it citing "expected behavior".

This is the heart of the matter:
> Shouldn't --sub-* options cover --sub-ass-* options in this case especially when their features are conflicted?

I'm expecting the answer is no. Imagine if mpv had these two options:
- `--sub-ass-font`
- `--sub-font`

I would expect the `--sub-ass-font` option that is specific to ASS subtitles to take presedence over the more general `--sub-font`.

If a user has gone to the trouble of setting [--sub-ass-style-overrides](https://mpv.io/manual/master/#options-sub-ass-style-overrides) then those are the styles they want applied to an ASS subtitle. In other words [--sub-ass-override](https://mpv.io/manual/master/#options-sub-ass-override) should only be about overriding the styles in the subtitle, it should not cause `-sub-ass-style-overrides` to be overridden.

**Comment by hooke007** [comment]
`sub-ass-override=yes` is for your use case. Isn't it?

**Comment by low-batt** [comment]
Not if you want some of the `--sub-*` settings in addition to the styles you specified with `--sub-ass-*` options. For that you have to set `sub-ass-override` to `force` and with that you currently loose some of the styles you specified with `--sub-ass-*` options.

**Comment by hooke007** [comment]
I think your actual request should be adding more levels in `sub-ass-override` such as `scale`, rather than breaking `force` that should do.

**Comment by low-batt** [comment]
Perfectly fine by me to just say this is the way it works and it is not worth breaking backward compatibility to change this. It does bother me that the description for `force` says "Like yes". The description for `yes` is "Apply all the --sub-ass-* style override options". As `force` is not applying _all_ the --sub-ass-* style override options it is really is not "Like yes".

**Comment by llyyr** [comment]
> I would be in favour of restoring old behavior to force. I used it to add a half transparent background under fonts to make them easier to read but I no longer can do it with updated mpv.

--sub-border-style=background-box
